src/files/glob-source.js

```diff
@@ -77,15 +107,29 @@ async function * toGlobSource ({ path, type, prefix }, options) {
 
   const globOptions = Object.assign({}, options.glob, {
     cwd: path,
-    nodir: true,
+    nodir: false,
     realpath: false,
     absolute: true
   })
 
   for await (const p of glob(path, '**/*', globOptions)) {
+    if (preserveMode || preserveMtime) {
+      const stat = await fs.stat(p)
+
+      if (options.preserveMode) {
+        mode = stat.mode
+      }
+
+      if (options.preserveMtime) {
+        mtime = parseInt(stat.mtimeMs / 1000)
+      }
+    }
+
     yield {
       path: toPosix(p.replace(prefix, '')),
-      content: fs.createReadStream(p)
+      content: fs.createReadStream(p),
+      mode,
+      mtime
```

Comment:
Directories get the same `mode` and `mtime` as files?

Reply:
They get nothing, unless `mode` or `mtime` are specified as options, in which case they that, unless `preserveMode` or `preserveMtime` are specified, in which case they get whatever they had on disk.

There was a bug here where files only got the `mode` or `mtime` of their containing folder but it is now squashed.